### PR TITLE
Install java version 21 in sle15sp6

### PR DESCRIPTION
**Pull request by mloviska**
java-11-openjdk is in legacy module for 15sp6.
Install L3 supported java version for each supported sles.

- Related ticket: [java-11 is installed from legacy module, update required java version in sle15 service packs](https://progress.opensuse.org/issues/154477)

#### Verification runs

* [sle-15-SP3-Server-DVD-Updates-x86_64-Build20240129](http://kepler.suse.cz/tests/22626#step/java/6)
* [sle-15-SP6-JeOS-for-kvm-and-xen-x86_64-Build4.22-jeos-base+sdk+desktop](http://kepler.suse.cz/tests/22614#step/java/12)
* [sle-12-SP3-Server-DVD-Updates-x86_64-Build20240129](http://kepler.suse.cz/tests/22619#step/java/6)
* [sle-15-SP4-Server-DVD-Updates-x86_64-Build20240129](http://kepler.suse.cz/tests/22622#step/java/6)
* [sle-12-SP5-Server-DVD-Updates-x86_64-Build20240129](http://kepler.suse.cz/tests/22625#step/java/6)
* [sle-15-SP5-Server-DVD-Updates-x86_64-Build20240129](http://kepler.suse.cz/tests/22623#step/java/6)
* [sle-15-SP1-Server-DVD-Updates-x86_64-Build20240129](http://kepler.suse.cz/tests/22624#step/java/6)
* [sle-15-SP2-Server-DVD-Updates-x86_64-Build20240129](http://kepler.suse.cz/tests/22627#step/java/6)

### CODE REVIEW

File: tests/console/java.pm [hunk 1]
```diff
@@ -46,10 +46,24 @@ sub run {
...
-    $cmd .= (is_sle('15+') || is_sle('=12-SP5') || is_leap) ? 'java-11-openjdk{,-demo,-devel} java-1_*' : 'java-*-devel';
+    my $pkgs_legacy = 'java-1_8*';
+
+    if (is_tumbleweed) {
+        $cmd .= 'java-*-devel';
+    } elsif (is_sle('15-SP6+') || is_leap('15.6+')) {
+        $cmd .= "java-21-openjdk{,-demo,-devel} $pkgs_legacy";
+    } elsif (is_sle('15+') || is_sle('=12-SP5') || is_leap) {
+        $cmd .= "java-11-openjdk{,-demo,-devel} $pkgs_legacy";
+    } elsif (is_sle('<12-SP5')) {
```
[the remark this answers is not in the document]
Comment (mloviska) [reply tests/console/java.pm:62]: 12sp5 is included in the condition above

[the remark this answers is not in the document]
Comment (m-dati) [reply tests/console/java.pm:62]: ok, I expected `<=`

[the remark this answers is not in the document]
Comment (m-dati) [reply tests/console/java.pm:62]: I am not sure the upper limit is included in that notation, but I trust you :)

**Review by m-dati — APPROVED** [review]

File: tests/console/java.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-dati) [reply tests/console/java.pm:62]: Sorry I didn't note the previous explicit check.

**Review by grisu48 — APPROVED** [review]
LGTM, but we have now a merge conflict that needs to be resolved.

**Pull request merged**